Cloudflare API token required for self-hosted version

#191 introduced the `CloudflareAPIKey` config variable as a mandatory config, which leads to the API not starting in our self-hosted environment.

Here is the thrown error message:
```
panic: invalid credentials: API Token must not be empty
```

Our current workaround is setting a dummy value in the API_CLOUDFLARE_API_TOKEN env var, which allows the API to start.

--- a/cmd/api/main.go
+++ b/cmd/api/main.go
@@ -118,9 +118,13 @@ func main() {
 		analyticsClient = analytics.New(config.SegmentWriteKey)
 	}
 
-	cloudflareClient, err := cloudflare.NewWithAPIToken(config.CloudflareAPIKey)
-	if err != nil {
-		panic(err)
+	var cloudflareClient *cloudflare.API
+	if config.CloudflareAPIKey != "" {
+		client, err := cloudflare.NewWithAPIToken(config.CloudflareAPIKey)
+		if err != nil {
+			panic(err)
+		}
+		cloudflareClient = client
 	}
 
 	connectPath, connectHandler := ssoreadyv1connect.NewSSOReadyServiceHandler(

